openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -644,87 +782,407 @@ def _add_to_existing_draft_change_log(
+    for ``U1`` in the change log, saying that it went from version 1 to version
+    1--i.e nothing about the Unit's defintion changed, but it was still affected
+    by other changes in the log.
+
+    Only call this function after all the records have already been created.
+
+    Note: The interface between ``DraftChangeLog`` and ``PublishLog`` is similar
+    enough that this function has been made to work with both.
+    """
+    branch_cls: type[Draft] | type[Published]
+    change_record_cls: type[DraftChangeLogRecord] | type[PublishLogRecord]
+    side_effect_cls: type[DraftSideEffect] | type[PublishSideEffect]
+    if isinstance(change_log, DraftChangeLog):
+        branch_cls = Draft
+        change_record_cls = DraftChangeLogRecord
+        side_effect_cls = DraftSideEffect
+        log_record_rel = "draft_log_record_id"
+    elif isinstance(change_log, PublishLog):
+        branch_cls = Published
+        change_record_cls = PublishLogRecord
+        side_effect_cls = PublishSideEffect
+        log_record_rel = "publish_log_record_id"
+    else:
+        raise TypeError(
+            f"expected DraftChangeLog or PublishLog, not {type(change_log)}"
+        )
+
+    # processed_entity_ids holds the entity IDs that we've already calculated
+    # side-effects for. This is to save us from recalculating side-effects for
+    # the same dependency relationships over and over again. So if we're calling
+    # this function in a loop for all the Components in a Unit, we won't be
+    # recalculating the Unit's side-effect on its Subsection, and its
+    # Subsection's side-effect on its Section each time through the loop.
+    # It also guards against infinite parent-child relationship loops, though
+    # those aren't *supposed* to happen anyhow.
     processed_entity_ids: set[int] = set()
-    for change in change_log.records.all():
-        _create_container_side_effects_for_draft_change(
-            change,
-            processed_entity_ids=processed_entity_ids,
+    for original_change in change_log.records.all():
+        affected_by_original_change = branch_cls.objects.filter(
+            version__dependencies=original_change.entity
         )
+        changes_and_affected = [
+            (original_change, current) for current in affected_by_original_change
+        ]
 
+        # These are the Published or Draft objects where we need to repoint the
+        # log_record (publish_log_record or draft_change_log) to point to the
+        # side-effect changes, e.g. the DraftChangeLogRecord that says, "This
+        # Unit's version stayed the same, but its dependency hash changed
+        # because a child Component's draft version was changed." We gather them
+        # all up in a list so we can do a bulk_update on them.
+        branch_objs_to_update_with_side_effects = []
+
+        while changes_and_affected:
+            original_change, affected = changes_and_affected.pop()
```

Comment:
I would rename this inner variable to something like `change` or `cause`, because:

* If we take "original change" to mean "a change coming directly from the change_log" (as is implied by `affected_by_original_change`), then I think anything at the 2nd layer of affect or lower is no longer "original".
* It's shadowing the outer loop's `original_change`, which is confusing.  I don't think it makes the code any more complicated for the inner loop (the one that descends the layers) to have its own variable.